src/Reflection/ReflectionClass.php

```diff
@@ -18,34 +20,39 @@ class ReflectionClass implements Reflection
     /**
      * @var NamespaceNode
      */
-    private $declaringNamespace;
+    private $declaringNamespace = null;
 
     /**
      * @var ReflectionMethod[]
      */
-    private $methods;
+    private $methods = [];
 
     /**
      * @var mixed[]
      */
-    private $constants;
+    private $constants = [];
 
     /**
      * @var ReflectionProperty[]
      */
-    private $properties;
+    private $properties = [];
 
     /**
      * @var string
      */
     private $filename;
 
-    private function __construct()
+    public function __construct($className = null)
     {
-        $this->declaringNamespace = null;
-        $this->methods = [];
-        $this->constants = [];
-        $this->properties = [];
+        if ($className) {
```

Comment:
I don't see another way around it, ideally I need two constructors.. one for internal (`::createFromNode`) use and one for public API :/